src/Models/HydrostaticFreeSurfaceModels/implicit_free_surface_solver.jl

```diff
@@ -0,0 +1,77 @@
+using Oceananigans.Solvers
+using Oceananigans.Operators
+
+struct ImplicitFreeSurfaceSolver{S}
+    solver   :: S
+end
+
+function ImplicitFreeSurfaceSolver(arch, template_field, 
+                                   vertically_integrated_lateral_face_areas;
+                                   Amatrix_operator=nothing, 
+                                   maxit=nothing, 
+                                   tol=nothing)
+       
+    ## This should probably end up in some operator and grid generic auxilliaries
+    ## NEED TO SORT OUT HALO FOR PERIODIC CASE
+    Ax_baro = vertically_integrated_lateral_face_areas.Ax
+    Ay_baro = vertically_integrated_lateral_face_areas.Ay
+    @inline Ax_∂xᶠᵃᵃ_baro(i, j, k, grid, c) = Ax_baro[i, j, 1] * ∂xᶠᵃᵃ(i, j, 1, grid, c)
+    @inline Ay_∂yᵃᶠᵃ_baro(i, j, k, grid, c) = Ay_baro[i, j, 1] * ∂yᵃᶠᵃ(i, j, 1, grid, c)
+    @inline function ∇²_baro(i, j, k, grid, c)
+       return  δxᶜᵃᵃ(i, j, 1, grid, Ax_∂xᶠᵃᵃ_baro, c) +
+               δyᵃᶜᵃ(i, j, 1, grid, Ay_∂yᵃᶠᵃ_baro, c)
+    end
```

Comment:
Would it be better to define this as an ordinary function, rather than a closure? Is there global data that we need to capture in the closure?

Reply:
I think something like Ax_baro or even Ax_∂xᶠᵃᵃ_baro might belong somewhere in Operators? Not sure it really belongs in solver?

Reply:
@glwagner I moved this to its own function, but it may really want to end up in some Operators foo.